### ISSUE TITLE
trivy config check for kubernetes (outdated-api) missing deprecation and wrong severity

### ISSUE THREAD

**Issue by bovy89** [aquasecurity/trivy#3294]
## What did you expect to happen?

1) Config check for kubernetes must take into account k8s version
2) Output below should report deprecation warning for PodSecurityPolicy and PodDisruptionBudget if tested against 1.23, and removed api error for PodSecurityPolicy and PodDisruptionBudget if tested against 1.25
3) Deprecated features should not rise an high level issue (low or medium is more appropriate)

## What happened instead?

1) Config check for kubernetes do not permit to set k8s target version
2) In both cases, PodSecurityPolicy is missing
3) Deprecated features rise an high level issue

## Output of run with `-debug`:

```cat > test.yaml <<'EOF'
---
apiVersion: policy/v1beta1
kind: PodDisruptionBudget
metadata:
  labels:
    gatekeeper.sh/system: "yes"
  name: gatekeeper-controller-manager
  namespace: gatekeeper-system
spec:
  minAvailable: 1
  selector:
    matchLabels:
      control-plane: controller-manager
      gatekeeper.sh/operation: webhook
      gatekeeper.sh/system: "yes"
---
apiVersion: policy/v1beta1
kind: PodSecurityPolicy
metadata:
  annotations:
    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
  labels:
    gatekeeper.sh/system: "yes"
  name: gatekeeper-admin
spec:
  allowPrivilegeEscalation: false
  fsGroup:
    ranges:
    - max: 65535
      min: 1
    rule: MustRunAs
  requiredDropCapabilities:
  - ALL
  runAsUser:
    rule: MustRunAsNonRoot
  seLinux:
    rule: RunAsAny
  supplementalGroups:
    ranges:
    - max: 65535
      min: 1
    rule: MustRunAs
  volumes:
  - configMap
  - projected
  - secret
  - downwardAPI
  - emptyDir

EOF

trivy config --exit-code 1 test.yaml --debug
2022-12-13T11:43:06.125+0100	DEBUG	Severities: ["UNKNOWN" "LOW" "MEDIUM" "HIGH" "CRITICAL"]
2022-12-13T11:43:06.164+0100	DEBUG	cache dir:  /Users/xxxx/Library/Caches/trivy
2022-12-13T11:43:06.164+0100	INFO	Misconfiguration scanning is enabled
2022-12-13T11:43:06.164+0100	DEBUG	Walk the file tree rooted at 'test.yaml' in parallel
2022-12-13T11:43:06.740+0100	DEBUG	OS is not detected.
2022-12-13T11:43:06.740+0100	INFO	Detected config files: 1
2022-12-13T11:43:06.740+0100	DEBUG	Scanned config file: test.yaml
2022-12-13T11:43:06.780+0100	DEBUG	Found an ignore file .trivyignore
2022-12-13T11:43:06.780+0100	DEBUG	These IDs will be ignored: ["KSV014" "KSV019" "KSV029" "KSV037" "KSV030" "KSV041" "KSV044" "KSV045" "KSV046" "KSV048" "KSV049" "KSV050" "KSV056" "KSV106"]

test.yaml (kubernetes)

Tests: 76 (SUCCESSES: 75, FAILURES: 1, EXCEPTIONS: 0)
Failures: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 0)

HIGH: apiVersion 'policy/v1beta1' and kind ‘PodDisruptionBudget' should be replaced with the new API 'policy.v1.PodDisruptionBudget'
See https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/api/policy/v1beta1/zz_generated.prerelease-lifecycle.go
══════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════════
apiVersion and kind has been deprecated
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
 test.yaml:2-6
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   2 ┌ apiVersion: policy/v1beta1
   3 │ kind: PodDisruptionBudget
   4 │ metadata:
   5 │   labels:
   6 └     gatekeeper.sh/system: "yes"
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

## Output of `trivy -v`:

```
Version: 0.35.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-11-08 12:09:11.842458865 +0000 UTC
  NextUpdate: 2022-11-08 18:09:11.842458665 +0000 UTC
  DownloadedAt: 2022-11-08 15:44:12.614908 +0000 UTC
```

**Comment by chen-keinan** [comment]
@bovy89 thank you for reporting 
- point 1 & 2 the depreciation evaluation today do not take under consideration the k8s version (will be fixed) 
- point 3 is already fixed and will be release with next `trivy v0.36.0` release all severities for will show as `Low`.

**Comment by knqyf263** [comment]
>point 1 & 2 the depreciation evaluation today do not take under consideration the k8s version (it will be fixed)

It is a feature enhancement. We created an issue for that. https://github.com/aquasecurity/trivy/issues/3338

>point 3 is already fixed and will be release with next trivy v0.36.0 release , all severity will show as Low.

It is already fixed in the main branch. The fix will be included in v0.36.0.
